imageengine/class-imagecdn.php

```diff
@@ -46,6 +47,24 @@ public function __construct() {
 		add_filter( 'plugin_action_links_' . IMAGE_CDN_BASE, array( self::class, 'add_action_link' ) );
 	}
 
+	/**
+	 * Add http headers for Client Hints, Feature Policy and Preconnect Resource Hint.
+	 */
+	public static function add_headers() {
+		// Add client hints.
+        header( 'Accept-CH: viewport-width, width, device-memory, dpr, rtt, downlink, ect' );
+
+		// Add resource hints and feature policy.
+		$options = self::get_options();
+		$host    = wp_parse_url( $options['url'], PHP_URL_HOST );
+		if ( ! empty( $host ) ) {
+    		$protocol = (is_ssl()) ? "https://" : "http://";
```

Comment:
We would need the port number to if we want to support non-standard ports (at least for the preconnect).  Does the feature-policy stuff need the port number?